src/components/UserProfile/FoodTicket.tsx

```diff
@@ -0,0 +1,171 @@
+"use client";
+
+import { generateClient } from "aws-amplify/api";
+import Image from "next/image";
+import { useEffect, useState } from "react";
+import { QRCode } from "react-qrcode-logo";
+
+import { type Schema } from "@/amplify/data/resource";
+import { createMessageAndCode } from "@/amplify/function/BusinessLogic/utils/crytography";
+import { getUpcomingFoodEventDetails } from "@/app/get-food-ticket/actions";
+import { useUser } from "@/components/contexts/UserContext";
+import { useQuery } from "@tanstack/react-query";
+
+import ProfileLinks from "./ProfileLinks";
+
+export default function UserFoodTicket() {
+  const client = generateClient<Schema>();
+
+  const userId = useUser().currentUser.userSub as string;
+
+  const { data, isFetching } = useQuery({
+    initialDataUpdatedAt: 0,
+    queryKey: ["User", userId],
+    queryFn: async () => {
+      const {
+        queuePosition,
+        eventName,
+        eventDescription,
+        eventTime,
+        timeslot,
+      } = await getUpcomingFoodEventDetails(userId);
+
+      let verificationCode = "";
+
+      const { data, errors } = await client.queries.GetUserMessageCode({
+        userMessage: userId,
+      });
+      if (errors) {
+        console.log(errors);
+        verificationCode = "Error when searching for your message code";
+        return;
+      }
+
+      if (data) {
+        const json = JSON.parse(data.body as string);
+        const code = json["value"];
+
+        const createdVerificationCode = createMessageAndCode(userId, code);
+        verificationCode = createdVerificationCode;
+      } else {
+        verificationCode = "Error when searching for your message code";
+      }
+
+      return {
+        queuePosition,
+        eventName,
+        eventDescription,
+        eventTime,
+        timeslot,
+        verificationCode,
+      };
+    },
+  });
+
+  // Not sure if we actually plan on showing their use code and i have no idea what its used for so for container purposes i will shorten it
```

Comment:
Maybe ideen can clear this up but I have no idea what the usercode does and it was messing with the borders of the container so i made this just to shorten it